### PR TITLE
Mimic the way scipy handles numpy in setup.py

### PR DESCRIPTION
**Pull request by blink1073**
Scipy has this figured out, let's just do what they do:
https://github.com/scipy/scipy/blob/master/setup.py

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -9,14 +9,14 @@
...
 
-DISTNAME            = 'scikit-image'
-DESCRIPTION         = 'Image processing routines for SciPy'
-LONG_DESCRIPTION    = descr
-MAINTAINER          = 'Stefan van der Walt'
-MAINTAINER_EMAIL    = 'stefan@sun.ac.za'
-URL                 = 'http://scikit-image.org'
-LICENSE             = 'Modified BSD'
-DOWNLOAD_URL        = 'http://github.com/scikit-image/scikit-image'
+DISTNAME = 'scikit-image'
```
Comment (blink1073) [inline setup.py]: Sorry, autopep8 got me here

File: setup.py [hunk 2]
```diff
@@ -41,16 +41,17 @@
 
 
 def configuration(parent_package='', top_path=None):
-    if os.path.exists('MANIFEST'): os.remove('MANIFEST')
+    if os.path.exists('MANIFEST'):
```
Comment (blink1073) [inline setup.py]: More autopep8 in this function.

File: setup.py [hunk 3]
```diff
@@ -59,17 +60,29 @@ def configuration(parent_package='', top_path=None):
 
 
 if __name__ == "__main__":
-    # purposely fail if numpy is not available
-    # other dependecies will be resolved by pip (install_requires)
     try:
         from numpy.distutils.core import setup
+        extra = {'configuration': configuration}
```
Comment (blink1073) [inline setup.py]: Start of substantive changes.

File: setup.py [hunk 4]
```diff
@@ -59,17 +60,29 @@ def configuration(parent_package='', top_path=None):
...
+                                                   'clean')):
+            # For these actions, NumPy is not required.
+            #
+            # They are required to succeed without Numpy for example when
+            # pip is used to install scikit-image when Numpy is not yet
+            # present in the system.
+            pass
+        else:
+            print('To install scikit-image from source, you will need numpy.\n' +
+                  'Install numpy with pip:\n' +
```
Comment (JDWarner) [inline setup.py]: Implicit string concatenation - no pluses needed!

I'd consider doubling the newlines on either side of the actual pip call.

**Comment by stefanv** [comment]
Sorry, @JDWarner, saw your comment now only--we can fix that in another commit, if you want.

**Comment by blink1073** [comment]
I'll fix @JDWarner's comment in a new PR, and I think we should exclude `scipy` and `matplotlib` when upgrading as well, thoughts?

**Comment by blink1073** [comment]
And also `PIL`, basically anything known to require compilation.

**Pull request merged**